src/query/functions/unconsolidated/timestamp.go

```diff
@@ -108,27 +111,27 @@ func (n *timestampNode) ProcessBlock(queryCtx *models.QueryContext, ID parser.No
 		return nil, err
 	}
 
-	if err = builder.AddCols(iter.StepCount()); err != nil {
+	count := iter.StepCount()
+	seriesCount := len(iter.SeriesMeta())
+	if err = builder.AddCols(count); err != nil {
 		return nil, err
 	}
 
+	bounds := b.Meta().Bounds
+	currentStep := float64(bounds.Start.Unix())
```

Comment:
Should the start be interpolated?  i.e. what if the start timestamp is half way through a second?

i.e. `currentStep := float64(bounds.Start.UnixNano()) / float64(time.Second)`

Reply:
Sorry, not quite sure what you mean by this, wouldn't it give you the same result?

Reply:
Just that half a second would get truncated because `Start.Unix()` returns seconds as int64 and loses the half second.